### PR TITLE
ContextMatcher: Allow matching on context baggage

### PR DESCRIPTION
**Pull request by abhinav**
**ContextMatcher: Allow matching on context baggage**

### CODE REVIEW

**Comment by abhinav** [comment]
@yarpc/golang

**Comment by kriskowal** [comment]
so say we all

**Comment by kriskowal** [comment]
should include a test that makes uses of the baggage matcher

**Comment by abhinav** [comment]
That's coming in the diff for which this functionality was added

File: transport/transporttest/context.go [hunk 1]
```diff
@@ -54,22 +78,32 @@ func NewContextMatcher(t *testing.T, ttl time.Duration) *ContextMatcher {
...
 
-	ttl := d.Sub(time.Now())
-	maxTTL := c.ttl + c.TTLDelta
-	minTTL := c.ttl - c.TTLDelta
-	if ttl > maxTTL || ttl < minTTL {
-		c.t.Logf("TTL out of expected bounds: %v < %v < %v", minTTL, ttl, maxTTL)
-		return false
+	if c.baggage != nil {
+		headers := baggage.FromContext(ctx)
+		if !assert.Equal(c.t, c.baggage, headers, "context baggage did not match") {
```
Comment (prashantv) [inline transport/transporttest/context.go]: we should be consistent in how we handle mismatches here. We do `assert.Equal` here, but above, we do a `t.Logf`. `assert.Equal` will fail the test, while `Logf` does not.

Comment (abhinav) [inline transport/transporttest/context.go]: 👍

Comment (abhinav) [inline transport/transporttest/context.go]: Fixed in https://github.com/yarpc/yarpc-go/pull/201

**Pull request merged**